presto-orc/src/main/java/com/facebook/presto/orc/OrcOutputBuffer.java

```diff
@@ -442,46 +439,46 @@ private void flushBufferToOutputStream()
 
     private void writeChunkToOutputStream(byte[] chunk, int offset, int length)
     {
-        if (compressor == null) {
+        if (compressor == null && !dwrfEncryptor.isPresent()) {
             compressedOutputStream.write(chunk, offset, length);
             return;
         }
 
-        checkArgument(length <= buffer.length, "Write chunk length must be less than compression buffer size");
-
-        int minCompressionBufferSize = compressor.maxCompressedLength(length);
-        if (compressionBuffer.length < minCompressionBufferSize) {
-            compressionBuffer = new byte[minCompressionBufferSize];
-        }
-
-        int compressedSize = compressor.compress(chunk, offset, length, compressionBuffer, 0, compressionBuffer.length);
-        if (compressedSize < length) {
-            if (dwrfEncryptor.isPresent()) {
-                compressionBuffer = dwrfEncryptor.get().encrypt(compressionBuffer, 0, compressedSize);
-                compressedSize = compressionBuffer.length;
+        if (compressor != null) {
+            int minCompressionBufferSize = compressor.maxCompressedLength(length);
+            if (compressionBuffer.length < minCompressionBufferSize) {
+                compressionBuffer = new byte[minCompressionBufferSize];
             }
-            int chunkHeader = (compressedSize << 1);
-            compressedOutputStream.write(chunkHeader & 0x00_00FF);
-            compressedOutputStream.write((chunkHeader & 0x00_FF00) >> 8);
-            compressedOutputStream.write((chunkHeader & 0xFF_0000) >> 16);
-            compressedOutputStream.writeBytes(compressionBuffer, 0, compressedSize);
-        }
-        else {
-            if (dwrfEncryptor.isPresent()) {
-                chunk = dwrfEncryptor.get().encrypt(chunk, 0, length);
-                length = chunk.length;
+            int compressedSize = compressor.compress(chunk, offset, length, compressionBuffer, 0, compressionBuffer.length);
+            if (compressedSize < length) {
+                if (dwrfEncryptor.isPresent()) {
+                    compressionBuffer = dwrfEncryptor.get().encrypt(compressionBuffer, 0, compressedSize);
+                    compressedSize = compressionBuffer.length;
```

Comment:
need to add check that the length after encryption does not exceed what the header can hold, ie 2^23